src/Workspaces/Core/Portable/Workspace/Solution/ProjectInfo.cs

```diff
@@ -380,7 +385,12 @@ internal class ProjectAttributes : IChecksummedObject, IObjectWritable
                 Id = id ?? throw new ArgumentNullException(nameof(id));
                 Name = name ?? throw new ArgumentNullException(nameof(name));
                 Language = language ?? throw new ArgumentNullException(nameof(language));
-                AssemblyName = assemblyName ?? throw new ArgumentNullException(nameof(assemblyName));
+
+                // Empty assembly names are not allowed.  For now just default to an unused assembly
+                // name.
+                AssemblyName = string.IsNullOrWhiteSpace(assemblyName)
+                    ? InvalidAssemblyNamePrefix + Interlocked.Increment(ref s_nextInvalidAssemblySuffix)
+                    : assemblyName;
```

Comment:
I think we should throw an exception if `language`, or `assemblyName` are null or the empty string. Maybe even `name` though the implications of that aren't clear to me.